src/app/query/stacks/balance/crypto-asset-balances.utils.ts

```diff
@@ -0,0 +1,139 @@
+import { getAssetStringParts } from '@stacks/ui-utils';
+import BigNumber from 'bignumber.js';
+import { FungibleTokenMetadata } from '@stacks/stacks-blockchain-api-types';
+
+import { isTransferableStacksFungibleTokenAsset } from '@app/common/crypto-assets/stacks-crypto-asset.utils';
+import type { AccountBalanceResponseBigNumber } from '@shared/models/account.model';
+import type {
+  StacksCryptoCurrencyAssetBalance,
+  StacksFungibleTokenAssetBalance,
+  StacksNonFungibleTokenAssetBalance,
+} from '@shared/models/crypto-asset-balance.model';
+import { createMoney } from '@shared/models/money.model';
+import { STX_DECIMALS } from '@shared/constants';
+
+export function createStacksCryptoCurrencyAssetTypeWrapper(
+  balance: BigNumber,
+  subBalance: BigNumber
+): StacksCryptoCurrencyAssetBalance {
+  return {
+    balance: createMoney(balance, 'STX'),
+    asset: {
+      blockchain: 'stacks',
+      decimals: STX_DECIMALS,
+      hasMemo: true,
+      name: 'Stacks',
+      symbol: 'STX',
+      type: 'crypto-currency',
+    },
+    subBalance: createMoney(subBalance, 'STX'),
+  };
+}
+
+function createStacksFtCryptoAssetBalanceTypeWrapper(
+  balance: BigNumber,
+  key: string
+): StacksFungibleTokenAssetBalance {
+  const { address, contractName, assetName } = getAssetStringParts(key);
+  return {
+    balance: createMoney(balance, ''),
+    asset: {
+      blockchain: 'stacks',
+      canTransfer: false,
+      contractAddress: address,
+      contractAssetName: assetName,
+      contractName,
+      decimals: 0,
+      hasMemo: false,
+      imageCanonicalUri: '',
+      name: '',
+      symbol: '',
+      type: 'fungible-token',
+    },
+    subBalance: createMoney(new BigNumber(0), ''),
+  };
+}
+
+function createStacksNftCryptoAssetBalanceTypeWrapper(
+  balance: BigNumber,
+  key: string
+): StacksNonFungibleTokenAssetBalance {
+  const { address, contractName, assetName } = getAssetStringParts(key);
```

Comment:
Suggest throwing if count isn't an integer

Reply:
I don't understand the comment here, do you mean BigNumber?

Reply:
```ts
const invalidNft = createStacksNftCryptoAssetBalanceTypeWrapper(new BigNumber(0.1), ''); // Error: NFT cannot be a non-interger
```

Something like this'd make sense, right? You can never have part of an NFT?